server.js

```diff
@@ -686,41 +769,7 @@ cache(function (data, match, sendBadge, request) {
 }));
 
 // AppVeyor CI integration.
-camp.route(/^\/appveyor\/ci\/([^\/]+\/[^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
-cache(function(data, match, sendBadge, request) {
-  var repo = match[1];  // eg, `gruntjs/grunt`.
-  var branch = match[2];
-  var format = match[3];
-  var apiUrl = 'https://ci.appveyor.com/api/projects/' + repo;
-  if (branch != null) {
-    apiUrl += '/branch/' + branch;
-  }
-  var badgeData = getBadgeData('build', data);
-  badgeData.logo = badgeData.logo || logos['appveyor'];
-  request(apiUrl, { headers: { 'Accept': 'application/json' } }, function(err, res, buffer) {
-    if (err != null) {
-      badgeData.text[1] = 'inaccessible';
-      sendBadge(format, badgeData);
-    }
-    try {
-      var data = JSON.parse(buffer);
-      var status = data.build.status;
-      if (status === 'success') {
-        badgeData.text[1] = 'passing';
-        badgeData.colorscheme = 'brightgreen';
-      } else if (status !== 'running' && status !== 'queued') {
-        badgeData.text[1] = 'failing';
-        badgeData.colorscheme = 'red';
-      } else {
-        badgeData.text[1] = status;
-      }
-      sendBadge(format, badgeData);
-    } catch(e) {
-      badgeData.text[1] = 'invalid';
-      sendBadge(format, badgeData);
-    }
-  });
-}));
+registerServices(require('./services/appveyor'));
```

Comment:
Eventually I imagine we'll have some sort of "auto registration" for all files in the `services` directory rather than manually doing it like this.

Reply:
This is pretty easy to do:

```js
glob.sync(`${__dirname}/services/*.js`).map(name => require(name)).forEach(registerServices);
```

Reply:
@paulmelnikow  The issue is that we also need to sort the URLs such that longer strings are registered before shorter ones, or enforce that URLs are not substrings of other URLs (and thus will never conflict). Right now there might be some implicit dependencies between badges based on the registration order :/

Reply:
Right… hmm, why don't we fix that as we're migrating the badges? Probably the few badges where this matters could go in the same file, so they can be kept in order relative to each other. `glob` will sort alphabetically, so in any case the results should be consistent.